app/api/chat/rocket_chat.py

```diff
@@ -24,76 +27,165 @@ def __init__(self, message, code=CODES.UNHANDLED, response=None) -> None:
+
+
+def generate_pass(size=10, chars=string.ascii_lowercase + string.digits):
+    return ''.join(random.choice(chars) for _ in range(size))
+
 
-    def login(method='login'):
-        res = requests.post(
-            login_url, json=dict(email=user.email, password=user.rocket_chat_password)
+def check_or_create_bot(event: Event, api_url='', login_url=''):
+    bot_email = 'openeventbot@openevent.com'
+    bot_user = User.query.filter_by(_email=bot_email).first_or_404()
+
+    if not (bot_user):
+        bot_user, _ = get_or_create(
+            User, _email=bot_email, _password=generate_pass(), first_name='openeventbot'
         )
-        data = res.json()
-        if res.status_code == 200:
-            token = data['data']['authToken']
-            save_token(token)
-            return dict(method=method, token=token, res=data)
-        else:
-            # Unhandled Case
-            logger.error('Error while rocket chat login: %s', data)
-            raise RocketChatException('Error while logging in', response=res)
-
-    def register(username_suffix=''):
-        register_url = api_url + '/api/v1/users.register'
-        register_data = {
-            'name': user.public_name or user.full_name,
-            'email': user.email,
-            'pass': user.rocket_chat_password,
-            'username': user.rocket_chat_username + username_suffix,
-        }
-        if registration_secret := settings['rocket_chat_registration_secret']:
-            register_data['secretURL'] = registration_secret
-
-        res = requests.post(register_url, json=register_data)
+        register(bot_user, event, api_url, login_url, '', True)
+
+    return bot_user
+
+
+def create_room(event: Event, user_id, api_url='', login_url=''):
+    bot_info = check_or_create_bot(event, api_url, login_url)
+    data = get_rocket_chat_token(bot_info, event, False, True)
+    bot_token = data['token']
+    bot_id = data['res']['data']['userId']
+
+    res = requests.post(
+        api_url + '/api/v1/groups.create',
+        json=dict(name=event.name + event.identifier, members=[bot_id]),
+        headers={
+            'X-Auth-Token': bot_token,
+            'X-User-Id': bot_id,
+        },
+    )
+    if not res.status_code == 200:
+        logger.error('Error while creating room : %s', res.json())
+        raise RocketChatException('Error while creating room', response=res)
+    else:
+        group_data = res.json()
+        event.chat_room_id = group_data['group']['_id']
+        db.session.add(event)
+        db.session.commit()
+        add_in_room(event, user_id, api_url, login_url)
+
+
+def add_in_room(event: Event, rocket_user_id, api_url='', login_url=''):
+    bot_info = check_or_create_bot(event, api_url, login_url)
+    data = get_rocket_chat_token(bot_info, event, False, True)
+    bot_token = data['token']
+    bot_id = data['res']['data']['userId']
+    room_info = {'roomId': event.chat_room_id, 'userId': rocket_user_id}
```

Comment:
chat room id is used here

Reply:
That's not frontend, no need to add something in schema which is not used in frontend. It gets exposed in the API

Reply:
Ok, but how will room id be given here then? as room id is needed to add a user into a room.

Reply:
Schema and Models are different things. Schema is for API, and model is for code. No change would be needed here if chat_room_id is not in the schema. Schema exposes stuff or takes a field as an input publicly through an API